tools/shared.py

```diff
@@ -1317,11 +1317,19 @@ def __getitem__(self, key):
     return self.instance()[key]
 
 
-Settings = SettingsManager()
+def verify_settings():
```

Comment:
Why not make this a method, then `Settings.verify()`?

Reply:
because the whole instance() thing.  The checking code would like self.instance()["WASM"] which just didn't looks right.  Maybe if we get rid of the singleton pattern we could to change this to method, but I don't want to do that in this PR